clang/lib/DPCT/AnalysisInfo.cpp

```diff
@@ -2831,6 +2847,101 @@ std::shared_ptr<MemVarInfo> MemVarInfo::buildMemVarInfo(const VarDecl *Var) {
   }
   return DpctGlobalInfo::getInstance().insertMemVarInfo(Var);
 }
+
+void MemVarInfo::migrateWithDeviceGlobal(const VarDecl *MemVar) {
+  auto &SM = DpctGlobalInfo::getSourceManager();
+  auto &Ctx = DpctGlobalInfo::getContext();
+  auto &MacroArgMap = DpctGlobalInfo::getMacroArgRecordMap();
+  auto TSI = MemVar->getTypeSourceInfo();
+  auto OriginTL = TSI->getTypeLoc();
+  auto TL = OriginTL;
+  auto BegLoc = MemVar->getBeginLoc();
+  if (BegLoc.isMacroID()) {
+    BegLoc = SM.getExpansionLoc(BegLoc);
+  }
+  auto LocInfo = DpctGlobalInfo::getLocInfo(BegLoc);
+  std::string Dims;
+  bool IsArray = OriginTL.getType()->isArrayType();
+  // 1.Remove bracket after var name
+  while (auto ATL = TL.getAs<clang::ArrayTypeLoc>()) {
+    auto BRange = ATL.getBracketsRange();
+    auto RT =
```

Comment:
spelling location of "[" and "]" might be separated and cause crash.
Can we try getDefinitionRange first?

getLocWithOffset(1) should be ok in most cases unless the user use some macro for "]"

Reply:
ok, will fix.